src/components/Auth/Profile/tabs/AddressesTab.jsx

```diff
@@ -1,93 +1,211 @@
+import { useEffect, useState } from "react";
+import InputCom from "../../../Helpers/InputCom"; // Your existing input component
+import { UserApi, updateUser } from "../../../../api/auth/user";
+
 export default function AddressesTab() {
+  const [user, setUser] = useState(null);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [addressData, setAddressData] = useState({
+    street: "",
+    city: "",
+    state: "",
+    postalCode: "",
+  });
+
+  useEffect(() => {
+    const fetchUser = async (id) => {
+      try {
+        const { data: userDetails } = await UserApi(id);
+        setUser(userDetails);
+      } catch (error) {
+        console.error("Error fetching user data:", error);
+      }
+    };
+
+    const token = localStorage.getItem("token");
+    if (!token) return;
+
+    try {
+      const payload = token.split(".")[1];
+      const decodedPayload = atob(payload);
+      const decodedPayloadObject = JSON.parse(decodedPayload);
+      fetchUser(decodedPayloadObject?.id);
+    } catch (error) {
+      console.error("Error decoding token:", error);
+    }
+  }, []);
+
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    setAddressData((prev) => ({ ...prev, [name]: value }));
+  };
+
+  const handleAddAddress = async () => {
+    if (!user) return;
+    const token = localStorage.getItem("token");
+
+    try {
+      const updatedAddresses = [...(user.addresses || []), addressData];
+      const updatedUser = { ...user, addresses: updatedAddresses };
+
+      await updateUser(user._id, token, updatedUser);
+
+      setUser(updatedUser);
+      setAddressData({ street: "", city: "", state: "", postalCode: "" });
+      setIsModalOpen(false);
+    } catch (error) {
+      console.error("Error adding address:", error);
```

Comment:
Do not print the error 
```suggestion
        enqueueSnackbar( "Something went wrong while Updating the user details , Please try again later!", { variant : "error" } );
```